Source/Extensions/UIViewExtensions.swift

```diff
@@ -68,3 +68,56 @@ extension UIView {
   }
 
 }
+
```

Comment:
Another idea that actually should be best for performance & maintainability (both in code & in sync on UI) would be to try `UIBezierPath`. 

It can draw completely custom paths. Using `move`, `addLine`, `addCurve` we should be able to completely custom draw all edges & corners, inside a system of 1 x 1. 

We then assign this path to a CAShapeLayer that we assign to the view, also keeping the reference inside the object. This will then allow us to scale it with the view as needed:
```swift
override func layoutSubviews {
    super.layoutSubviews()

    customBorderLayer.frame = bounds;
}
```